lib/plugins/aws/package/compile/functions/index.test.js

```diff
@@ -71,6 +71,211 @@ describe('AwsCompileFunctions', () => {
+            cliArgs: ['package'],
+            awsRequestStubMap: {
+              CloudFormation: {
+                describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+              },
+            },
+          });
+        })
+        .then(data => {
+          fs.renameSync(originalLayer, sourceChangeLayer);
+          fs.renameSync(backupLayer, originalLayer);
+          expect(firstVersionId).to.not.equal(
+            data.cfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref
+          );
+        });
+    });
+
+    it('should create same version ids for identical lambdas with identical layer zip content', () => {
+      const firstVersionId = firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref;
+
+      const functionLayersBasePath = fixtures.map.functionLayers;
+      const originalLayer = path.join(functionLayersBasePath, 'testLayer');
+      const sourceChangeLayer = path.join(
+        functionLayersBasePath,
+        'extra_layers',
+        'testLayerDuplicate'
+      );
+      const backupLayer = path.join(functionLayersBasePath, 'extra_layers', 'testLayerBackup');
+
+      fs.renameSync(originalLayer, backupLayer);
+      fs.renameSync(sourceChangeLayer, originalLayer);
+      return fixtures
+        .extend('functionLayers', {})
+        .then(fixturePath => {
+          return runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+            awsRequestStubMap: {
+              CloudFormation: {
+                describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+              },
+            },
+          });
+        })
+        .then(data => {
+          fs.renameSync(originalLayer, sourceChangeLayer);
+          fs.renameSync(backupLayer, originalLayer);
+          expect(firstVersionId).to.equal(
+            data.cfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref
+          );
+        });
+    });
+
+    it('should create different version ids for identical lambdas with different layer properties', () => {
+      const firstVersionId = firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref;
+
+      return fixtures
+        .extend('functionLayers', {
+          layers: { testLayer: { path: 'testLayer', description: 'Different description' } },
+        })
+        .then(fixturePath => {
+          return runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+            awsRequestStubMap: {
+              CloudFormation: {
+                describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+              },
+            },
+          });
+        })
+        .then(data => {
+          expect(firstVersionId).to.not.equal(
+            data.cfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref
+          );
+        });
+    });
+
+    it('should create the same version ids for identical lambdas with layers with only different S3Key paths', () => {
```

Comment:
`should ignore changing character of S3Key paths when generating layer version id`